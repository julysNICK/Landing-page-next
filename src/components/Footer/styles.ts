import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../Textcomponent/styles';
import { Container as SectionContainer } from '../SectionContainer/styles';

export const Container = styled.footer`
  ${({ theme }) => css`
    text-align: center;
    border-top: 0.1rem solid ${theme.colors.mediumGray};
    a {
      color: inherit;
      text-decoration: none;
    }
    & ${TextComponent} {
      font-size: ${theme.font.sizes.small};
    }
    & ${SectionContainer} {
      padding-top: 0;
      padding-bottom: 0;
    }
  `}
`;
