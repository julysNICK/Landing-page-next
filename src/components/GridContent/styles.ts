import styled, { css } from 'styled-components';

export const Container = styled.div`
  text-align: center;
  max-width: 58rem;
  margin: 0 auto;
`;

export const Html = styled.div`
  ${({ theme }) => css`
    margin: ${theme.spacings.xhuge} 0;
  `}
`;
