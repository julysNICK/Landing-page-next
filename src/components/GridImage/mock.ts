import { GridImageProps } from '.';

export default {
  background: false,
  title: 'My grid',
  description:
    'Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde',
  grid: [
    {
      altText: 'Teste 1',
      srcImg: 'https://source.unsplash.com/random/800x800?r=1',
    },
    {
      altText: 'Teste 2',
      srcImg: 'https://source.unsplash.com/random/800x800?r=2',
    },
    {
      altText: 'Teste 3',
      srcImg: 'https://source.unsplash.com/random/800x800?r=3',
    },
    {
      altText: 'Teste 4',
      srcImg: 'https://source.unsplash.com/random/800x800?r=4',
    },
    {
      altText: 'Teste 5',
      srcImg: 'https://source.unsplash.com/random/800x800?r=5',
    },
    {
      altText: 'Teste 6',
      srcImg: 'https://source.unsplash.com/random/800x800?r=6',
    },
  ],
} as GridImageProps;
