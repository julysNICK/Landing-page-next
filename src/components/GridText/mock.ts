import { GridTextProps } from '.';

export default {
  background: false,
  title: 'My grid',
  description:
    'Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde',
  grid: [
    {
      title: 'Teste 1',
      description:
        'Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.',
    },
    {
      title: 'Teste 2',
      description:
        'Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.',
    },
    {
      title: 'Teste 3',
      description:
        'Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis cum delectus molestias. Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde, reprehenderit minus ad dolores provident maiores.',
    },
  ],
} as GridTextProps;
