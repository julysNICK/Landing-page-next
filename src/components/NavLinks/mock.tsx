export default [
  {
    children: 'Link 1',
    link: '#target1',
    newTab: false,
  },
  {
    children: 'Link 2',
    link: '#target2',
    newTab: false,
  },
  {
    children: 'Link 3',
    link: '#target3',
    newTab: false,
  },
  {
    children: 'Link 4',
    link: '#target4',
    newTab: false,
  },
  {
    children: 'Link 5',
    link: '#target5',
    newTab: false,
  },
  {
    children: 'Link 6',
    link: '#target6',
    newTab: false,
  },
  {
    children: 'Link 7',
    link: '#target7',
    newTab: false,
  },
  {
    children: 'Link 8',
    link: '#target8',
    newTab: false,
  },
  {
    children: 'Link 9',
    link: '#target9',
    newTab: false,
  },
  {
    children: 'Link 10',
    link: '#target10',
    newTab: false,
  },
];
