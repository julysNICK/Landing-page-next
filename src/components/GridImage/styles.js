import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';
import { Container as TextComponent } from '../Textcomponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    > ${TextComponent} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};
    @media ${theme.font.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    overflow: hidden;
  `}
`;

export const Image = styled.img`
  ${({ theme }) => css`
    width: 100%;
    transition: all 300ms ease-in-out;
    &:hover {
      transform: scale(1.2) rotate(10deg);
    }
  `}
`;
