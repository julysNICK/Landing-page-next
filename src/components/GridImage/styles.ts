import styled, { css } from 'styled-components';

import { Container as TextComponent } from '../Textcomponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    > ${TextComponent} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};
    @media ${theme.font.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridElement = styled.div`
  overflow: hidden;
`;

export const Image = styled.img`
  width: 100%;
  transition: all 300ms ease-in-out;
  &:hover {
    transform: scale(1.2) rotate(10deg);
  }
`;
