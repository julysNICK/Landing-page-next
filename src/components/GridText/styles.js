import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';
import { Container as TextComponent } from '../Textcomponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    ${TextComponent} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};
    overflow: hidden;
    width: 100%;
    @media ${theme.font.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    ${HeadingContainer} {
      position: relative;
      left: 5rem;
    }
    ${HeadingContainer}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: 7rem;
      top: -3rem;
      left: -5rem;
      transform: rotate(5deg);
    }
  `}
`;
