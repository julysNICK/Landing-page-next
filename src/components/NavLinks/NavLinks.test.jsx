import { renderTheme } from '../../styles/render-theme';
import { NavLinks } from '.';
import mock from './mock';
import { screen } from '@testing-library/dom';
import { theme } from '../../styles/theme';
describe('<NavLinks />', () => {
  it('should render links', () => {
    renderTheme(<NavLinks links={mock} />);

    expect(screen.getAllByRole('link')).toHaveLength(mock.length);
  });
  it('should not render links', () => {
    renderTheme(<NavLinks />);

    expect(screen.queryAllByText(/links/i)).toHaveLength(0);
  });
  it('should render links', () => {
    renderTheme(<NavLinks links={mock} />);

    expect(screen.getByText(/link 10/i).parentElement).toHaveStyleRule(
      'flex-flow',
      'column wrap',
      {
        media: theme.font.media.lteMedium,
      },
    );
  });
  it('should  match snapshot', () => {
    const { container } = renderTheme(<NavLinks links={mock} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
