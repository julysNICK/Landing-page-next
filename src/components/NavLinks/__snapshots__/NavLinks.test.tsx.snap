// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavLinks /> should  match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
}

.c1 {
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.6rem;
  padding: 1.6rem;
  color: #0A1128;
  position: relative;
}

.c1::after {
  content: '';
  position: absolute;
  bottom: 0.76rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background: #dc143c;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c1:hover::after {
  left: 25%;
  width: 50%;
}

@media (max-width:768px) {
  .c0 {
    -webkit-flex-flow: column wrap;
    -ms-flex-flow: column wrap;
    flex-flow: column wrap;
    -webkit-align-content: center;
    -ms-flex-line-pack: center;
    align-content: center;
  }
}

<nav
  class="c0"
>
  <a
    class="c1"
    href="#target1"
    target="_self"
  >
    Link 1
  </a>
  <a
    class="c1"
    href="#target2"
    target="_self"
  >
    Link 2
  </a>
  <a
    class="c1"
    href="#target3"
    target="_self"
  >
    Link 3
  </a>
  <a
    class="c1"
    href="#target4"
    target="_self"
  >
    Link 4
  </a>
  <a
    class="c1"
    href="#target5"
    target="_self"
  >
    Link 5
  </a>
  <a
    class="c1"
    href="#target6"
    target="_self"
  >
    Link 6
  </a>
  <a
    class="c1"
    href="#target7"
    target="_self"
  >
    Link 7
  </a>
  <a
    class="c1"
    href="#target8"
    target="_self"
  >
    Link 8
  </a>
  <a
    class="c1"
    href="#target9"
    target="_self"
  >
    Link 9
  </a>
  <a
    class="c1"
    href="#target10"
    target="_self"
  >
    Link 10
  </a>
</nav>
`;
